azurerm/internal/services/domainservices/active_directory_domain_service_test.go

```diff
@@ -0,0 +1,531 @@
+)
+
+type ActiveDirectoryDomainServiceResource struct {
+	adminPassword string
+}
+
+type ActiveDirectoryDomainServiceReplicaSetResource struct {
+}
+
+// Running all tests sequentially here in the same acctest (including the data source) since you can create only one
+// AADDS resource per tenant, or per location, or per subscription, making parallel testing infeasible.
+func TestAccActiveDirectoryDomainService(t *testing.T) {
```

Comment:
could we break this up into test functions and then sequentially run them as we do elsewhere in the provider so its a bit mroe clear what test is what?

Reply:
This could definitely be explained better; I've added a more thorough comment rationalising the test, and a TODO that we should revisit this in future when the service is better optimised